### DISCUSSION TITLE
怎么避免某个ip被加入到ipset？

### DISCUSSION THREAD

**Discussion by PHCSJC** [IrineSistiana/mosdns#341] [Q&A] [answered]
我的vps ip被加入到ipset里了，导致代理不通，如何避免这个ip被加入到ipset里呢？

场景是这样：vps是代理也是我的网站（所以有域名），而这个域名刚好在geosite的已知国外域名中，导致我自己只要一访问自己的网站，ip就被加到ipset里了，所以我想把我的ip排除在外，怎么排除呢？

另外，我不想通过域名的方式来排除，因为域名不固定，所以用ip来排除最合适，请问如何排除呢？感谢！

**Comment by PHCSJC** [comment]
也就是这个地方怎么修改呢？里面还可以嵌套if吗？如果能嵌套，是不是，当不执行ipset，同时也不执行_return了？

        - if:
            - query_is_non_local_domain  # 已知的非本地域名
          exec:
            - forward_remote             # 用远程服务器
            - ipset
            - _return

**Comment by xcf13363175** [comment]
之前也碰到了这个问题，加个host可解决

**Comment by PHCSJC** [comment]
> 之前也碰到了这个问题，加个host可解决

你是指“ip 域名”这样的吗？如果域名不固定怎么办呢？

**Comment by xcf13363175** [comment]
可以自己做一个白名单，里面内容走local

**Comment by PHCSJC** [comment]
@xcf13363175 能分享下配置吗？实在不会，非常感谢！

**Comment by IrineSistiana** [comment, marked as answer]
执行序列支持嵌套。用 if 让这个 IP 跳过 ipset。

**Comment by PHCSJC** [comment]
你好，我试了下，还是不行，大佬看看是哪里不对？下面是相关的配置

        - if:
            - query_is_non_local_domain  # 已知的非本地域名
          exec:
            - forward_remote             # 用远程服务器
            - if:
                - '!response_has_local_ip'  # 过滤掉非本地的 IP。
              exec:
            - ipset
            - _return





  # 匹配本地 IP 的插件
  - tag: response_has_local_ip
    type: response_matcher
    args:
      ip:
        - 'ext:./geoip.dat:cn'
        - 'ext:./vpsip'



cat vpsip
1.2.3.4/32


当执行域名查询时，1.2.3.4还是会被加到ipset里，如果把- ipset注释掉，就不会加入，说明的确是在那块加入的

**Reply by IrineSistiana** [reply]
缩进错了

**Comment by PHCSJC** [comment]
可以了，非常感谢！
